In [189]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [190]:
graffiti = pd.read_csv("311_Service_Requests_-_Graffiti_Removal.csv")

In [191]:
graffiti

Creation Date           Status Completion Date Service Request Number  \
0         01/01/2011  Completed - Dup      01/03/2011            11-00002268   
1         01/01/2011  Completed - Dup      01/03/2011            11-00002378   
2         01/01/2011  Completed - Dup      01/03/2011            11-00002391   
3         01/01/2011  Completed - Dup      01/03/2011            11-00002639   
4         01/01/2011  Completed - Dup      01/04/2011            11-00001746   
5         01/01/2011  Completed - Dup      01/05/2011            11-00003283   
6         01/01/2011  Completed - Dup      01/12/2011            11-00002866   
7         01/01/2011        Completed      01/03/2011            11-00001286   
8         01/01/2011        Completed      01/03/2011            11-00001666   
9         01/01/2011        Completed      01/03/2011            11-00001751   
10        01/01/2011        Completed      01/03/2011            11-00001870   
11        01/01/2011        Completed      01/03/2011            11-00002200   
12        01/01/2011        Completed      01/03/2011            11-00002206   
13        01/01/2011        Completed      01/03/2011            11-00002234   
14        01/01/2011        Completed      01/03/2011            11-00002237   
15        01/01/2011        Completed      01/03/2011            11-00002267   
16        01/01/2011        Completed      01/03/2011            11-00002365   
17        01/01/2011        Completed      01/03/2011            11-00002547   
18        01/01/2011        Completed      01/03/2011            11-00002626   
19        01/01/2011        Completed      01/03/2011            11-00002678   
20        01/01/2011        Completed      01/03/2011            11-00002679   
21        01/01/2011        Completed      01/03/2011            11-00002681   
22        01/01/2011        Completed      01/03/2011            11-00002709   
23        01/01/2011        Completed      01/03/2011            11-00002945   
24        01/01/2011        Completed      01/03/2011            11-00003386   
25        01/01/2011        Completed      01/04/2011            11-00002071   
26        01/01/2011        Completed      01/04/2011            11-00002549   
27        01/01/2011        Completed      01/04/2011            11-00002599   
28        01/01/2011        Completed      01/04/2011            11-00002605   
29        01/01/2011        Completed      01/04/2011            11-00002608   
...              ...              ...             ...                    ...   
855153    03/29/2017        Completed      03/29/2017            17-01761965   
855154    03/29/2017        Completed      03/29/2017            17-01761984   
855155    03/29/2017        Completed      03/29/2017            17-01762001   
855156    03/29/2017        Completed      03/29/2017            17-01762003   
855157    03/29/2017        Completed      03/29/2017            17-01762008   
855158    03/29/2017        Completed      03/29/2017            17-01762009   
855159    03/29/2017        Completed      03/29/2017            17-01762020   
855160    03/29/2017        Completed      03/29/2017            17-01762034   
855161    03/29/2017        Completed      03/29/2017            17-01762065   
855162    03/29/2017        Completed      03/29/2017            17-01762096   
855163    03/29/2017        Completed      03/29/2017            17-01762121   
855164    03/29/2017        Completed      03/29/2017            17-01762129   
855165    03/29/2017        Completed      03/29/2017            17-01762132   
855166    03/29/2017        Completed      03/29/2017            17-01762146   
855167    03/29/2017        Completed      03/29/2017            17-01762183   
855168    03/29/2017        Completed      03/29/2017            17-01762208   
855169    03/29/2017        Completed      03/29/2017            17-01762232   
855170    03/29/2017        Completed      03/29/2017            17-01762255   
855171 

In [192]:
graffiti["Type of Service Request"].value_counts()

Graffiti Removal    855183
Name: Type of Service Request, dtype: int64

In [193]:
graffiti.shape

(855183, 18)

In [194]:
sum(pd.isnull(graffiti["SSA"]))

650709

In [195]:
650709/855183

0.7609002985325948

We can see that the type fo service rquest is identical for the entire dataframe. As such, we will drop the service request number column as it doesnt tll us anything interesting. We will also drop the street address column as our anaysis will be focused on Zip Codes. Since we will not be perfoming any form of geociding, I will also drop the columns correspinding the the X and Y co-ordinates as well as the latiude and the longitude and the location.

From the above we can see that around 76% of data is not reported in the SSA column - thus the SSA column will also be dropped.

In [196]:
graffiti.drop('Service Request Number', axis=1, inplace=True)
graffiti.drop("Street Address", axis=1, inplace=True)
graffiti.drop("X Coordinate", axis=1, inplace=True)
graffiti.drop("Y Coordinate", axis=1, inplace=True)
graffiti.drop("Latitude", axis=1, inplace=True)
graffiti.drop("Longitude", axis=1, inplace=True)
graffiti.drop("SSA", axis=1, inplace=True)
graffiti.drop("Location", axis=1, inplace=True)

In [197]:
graffiti

Creation Date           Status Completion Date Type of Service Request  \
0         01/01/2011  Completed - Dup      01/03/2011        Graffiti Removal   
1         01/01/2011  Completed - Dup      01/03/2011        Graffiti Removal   
2         01/01/2011  Completed - Dup      01/03/2011        Graffiti Removal   
3         01/01/2011  Completed - Dup      01/03/2011        Graffiti Removal   
4         01/01/2011  Completed - Dup      01/04/2011        Graffiti Removal   
5         01/01/2011  Completed - Dup      01/05/2011        Graffiti Removal   
6         01/01/2011  Completed - Dup      01/12/2011        Graffiti Removal   
7         01/01/2011        Completed      01/03/2011        Graffiti Removal   
8         01/01/2011        Completed      01/03/2011        Graffiti Removal   
9         01/01/2011        Completed      01/03/2011        Graffiti Removal   
10        01/01/2011        Completed      01/03/2011        Graffiti Removal   
11        01/01/2011        Completed      01/03/2011        Graffiti Removal   
12        01/01/2011        Completed      01/03/2011        Graffiti Removal   
13        01/01/2011        Completed      01/03/2011        Graffiti Removal   
14        01/01/2011        Completed      01/03/2011        Graffiti Removal   
15        01/01/2011        Completed      01/03/2011        Graffiti Removal   
16        01/01/2011        Completed      01/03/2011        Graffiti Removal   
17        01/01/2011        Completed      01/03/2011        Graffiti Removal   
18        01/01/2011        Completed      01/03/2011        Graffiti Removal   
19        01/01/2011        Completed      01/03/2011        Graffiti Removal   
20        01/01/2011        Completed      01/03/2011        Graffiti Removal   
21        01/01/2011        Completed      01/03/2011        Graffiti Removal   
22        01/01/2011        Completed      01/03/2011        Graffiti Removal   
23        01/01/2011        Completed      01/03/2011        Graffiti Removal   
24        01/01/2011        Completed      01/03/2011        Graffiti Removal   
25        01/01/2011        Completed      01/04/2011        Graffiti Removal   
26        01/01/2011        Completed      01/04/2011        Graffiti Removal   
27        01/01/2011        Completed      01/04/2011        Graffiti Removal   
28        01/01/2011        Completed      01/04/2011        Graffiti Removal   
29        01/01/2011        Completed      01/04/2011        Graffiti Removal   
...              ...              ...             ...                     ...   
855153    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855154    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855155    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855156    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855157    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855158    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855159    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855160    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855161    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855162    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855163    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855164    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855165    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855166    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855167    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855168    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855169    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855170    03/29/2017        Completed 

In [198]:
from datetime import datetime

def days_between(d1, d2):
    d1 = datetime.strptime(d1, "%m/%d/%Y")
    d2 = datetime.strptime(d2, "%m/%d/%Y")
    return abs((d2 - d1).days)
#obtained from stack overflow

In [199]:
a = list(graffiti['Creation Date'])
b = list(graffiti["Completion Date"])

combined = list(zip(a,b))



In [200]:
days_list = []

for i in combined:
    start, end = i
    if type(start) and type(end) == str:
        days = days_between(start, end)
        days_list.append(days)
# we are ensuring is is of type str as we have some nan which are of type float.

In [201]:
len(days_list)

854960

In [202]:
graffiti.shape

(855183, 10)

In [203]:
855183-854960

223

223 cases where work has yet to be completed for graffiti. This is surprisngly small - looks like chicago has a good hold on taking care of graffiti issues.

In [204]:
df_days = pd.DataFrame({"Response Time (Days)": days_list})

In [205]:
graffiti = pd.concat([graffiti,df_days], axis=1)

In [206]:
graffiti

Creation Date           Status Completion Date Type of Service Request  \
0         01/01/2011  Completed - Dup      01/03/2011        Graffiti Removal   
1         01/01/2011  Completed - Dup      01/03/2011        Graffiti Removal   
2         01/01/2011  Completed - Dup      01/03/2011        Graffiti Removal   
3         01/01/2011  Completed - Dup      01/03/2011        Graffiti Removal   
4         01/01/2011  Completed - Dup      01/04/2011        Graffiti Removal   
5         01/01/2011  Completed - Dup      01/05/2011        Graffiti Removal   
6         01/01/2011  Completed - Dup      01/12/2011        Graffiti Removal   
7         01/01/2011        Completed      01/03/2011        Graffiti Removal   
8         01/01/2011        Completed      01/03/2011        Graffiti Removal   
9         01/01/2011        Completed      01/03/2011        Graffiti Removal   
10        01/01/2011        Completed      01/03/2011        Graffiti Removal   
11        01/01/2011        Completed      01/03/2011        Graffiti Removal   
12        01/01/2011        Completed      01/03/2011        Graffiti Removal   
13        01/01/2011        Completed      01/03/2011        Graffiti Removal   
14        01/01/2011        Completed      01/03/2011        Graffiti Removal   
15        01/01/2011        Completed      01/03/2011        Graffiti Removal   
16        01/01/2011        Completed      01/03/2011        Graffiti Removal   
17        01/01/2011        Completed      01/03/2011        Graffiti Removal   
18        01/01/2011        Completed      01/03/2011        Graffiti Removal   
19        01/01/2011        Completed      01/03/2011        Graffiti Removal   
20        01/01/2011        Completed      01/03/2011        Graffiti Removal   
21        01/01/2011        Completed      01/03/2011        Graffiti Removal   
22        01/01/2011        Completed      01/03/2011        Graffiti Removal   
23        01/01/2011        Completed      01/03/2011        Graffiti Removal   
24        01/01/2011        Completed      01/03/2011        Graffiti Removal   
25        01/01/2011        Completed      01/04/2011        Graffiti Removal   
26        01/01/2011        Completed      01/04/2011        Graffiti Removal   
27        01/01/2011        Completed      01/04/2011        Graffiti Removal   
28        01/01/2011        Completed      01/04/2011        Graffiti Removal   
29        01/01/2011        Completed      01/04/2011        Graffiti Removal   
...              ...              ...             ...                     ...   
855153    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855154    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855155    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855156    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855157    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855158    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855159    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855160    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855161    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855162    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855163    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855164    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855165    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855166    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855167    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855168    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855169    03/29/2017        Completed      03/29/2017        Graffiti Removal   
855170    03/29/2017        Completed 

In [207]:
graffiti['Response Time (Days)'].groupby(graffiti['ZIP Code']).describe().unstack()

count       mean          std  min  25%  50%  75%      max
ZIP Code                                                              
0.0         161.0   7.857143    23.034718  0.0  0.0  0.0  0.0    129.0
60601.0    4168.0   2.548944    51.599104  0.0  0.0  0.0  1.0   2348.0
60602.0    1201.0   1.721066     6.173030  0.0  0.0  0.0  1.0     98.0
60603.0     915.0   1.433880     6.589790  0.0  0.0  0.0  1.0    145.0
60604.0     746.0   1.446381     4.465281  0.0  0.0  0.0  1.0     49.0
60605.0    5124.0   2.233802     5.800138  0.0  0.0  0.0  2.0     87.0
60606.0    2072.0   1.541506     6.065967  0.0  0.0  0.0  1.0    182.0
60607.0    6816.0   3.716696    10.878775  0.0  0.0  1.0  2.0    182.0
60608.0   73958.0   4.962033   167.076075  0.0  0.0  1.0  3.0  32071.0
60609.0   38722.0   4.547208   188.498675  0.0  0.0  0.0  2.0  26186.0
60610.0    4228.0   5.150189    19.378672  0.0  0.0  1.0  3.0    411.0
60611.0    3084.0   2.085603     5.889109  0.0  0.0  0.0  1.0     72.0
60612.0   11427.0   5.321782    22.132914  0.0  0.0  1.0  3.0   1561.0
60613.0   16245.0   4.613912    46.214910  0.0  0.0  1.0  3.0   3206.0
60614.0   16812.0  37.764870   989.475948  0.0  0.0  1.0  3.0  30274.0
60615.0    1536.0   4.018229     8.155152  0.0  0.0  1.0  3.0     60.0
60616.0   10526.0   4.677180    34.308273  0.0  0.0  1.0  3.0   2388.0
60617.0   19570.0   4.939295    25.034299  0.0  0.0  1.0  4.0   1126.0
60618.0   62343.0   7.199589   220.311851  0.0  0.0  1.0  4.0  31163.0
60619.0    1488.0   4.988575    10.043108  0.0  0.0  1.0  5.0    105.0
60620.0    1967.0   4.287748     9.619652  0.0  0.0  0.0  3.0    135.0
60621.0    1711.0  18.617767   633.049344  0.0  0.0  0.0  1.0  26186.0
60622.0   35243.0   4.672956    17.522595  0.0  0.0  0.0  3.0    801.0
60623.0   51457.0  13.946616   528.383720  0.0  0.0  1.0  3.0  32071.0
60624.0    2897.0   4.395927    15.671799  0.0  0.0  0.0  2.0    301.0
60625.0   35593.0   9.735566   361.764257  0.0  0.0  1.0  3.0  22741.0
60626.0   10596.0   7.391091    24.647306  0.0  0.0  1.0  6.0    458.0
60627.0       1.0   1.000000          NaN  1.0  1.0  1.0  1.0      1.0
60628.0    2785.0   4.409695     8.265519  0.0  0.0  1.0  4.0     56.0
60629.0   49571.0  20.976519   740.023860  0.0  0.0  1.0  3.0  32401.0
...           ...        ...          ...  ...  ...  ...  ...      ...
60632.0   94955.0   8.773419   405.693085  0.0  0.0  1.0  2.0  32401.0
60633.0    1575.0   3.760635     7.428813  0.0  0.0  1.0  4.0     66.0
60634.0   16233.0   5.135958    19.104672  0.0  0.0  1.0  4.0    464.0
60635.0       5.0   5.800000    10.802777  0.0  0.0  1.0  3.0     25.0
60636.0    3195.0   3.158372     8.538980  0.0  0.0  0.0  1.0     79.0
60637.0    1748.0   3.797483     7.982892  0.0  0.0  1.0  3.0     65.0
60638.0    9770.0   3.210542     6.455165  0.0  0.0  1.0  3.0     87.0
60639.0   40150.0   6.106102   214.262476  0.0  0.0  1.0  4.0  30274.0
60640.0   14980.0   4.735247    17.787971  0.0  0.0  1.0  3.0    852.0
60641.0   32726.0   4.923761    13.926715  0.0  0.0  1.0  4.0    427.0
60642.0   12145.0   5.492384    13.409334  0.0  0.0  1.0  4.0    357.0
60643.0     933.0  72.721329  1496.286274  0.0  0.0  0.0  3.0  32401.0
60644.0    3172.0   4.908260    12.640361  0.0  0.0  1.0  3.0    202.0
60645.0    8258.0   9.114677   251.033846  0.0  0.0  1.0  6.0  22741.0
60646.0    1569.0   5.328235    10.868003  0.0  0.0  1.0  5.0    137.0
60647.0   66920.0   4.891811    22.033888  0.0  0.0  1.0  3.0   3882.0
60649.0    1533.0   4.864971     8.872577  0.0  0.0  1.0  5.0     96.0
60651.0    7598.0   5.067518    12.821957  0.0  0.0  1.0  4.0    406.0
60652.0    7969.0   2.613502     5.844964  0.0  0.0  1.0  3.0    142.0
60653.0     967.0   3.583247     9.798331  0.0  0.0  0.0  2.0    182.0
60654.0    4580.0   2.152838     7.074727  0.0  0.0  0.0  1.0    141.0
60655.0     842.0   1.789786     5.265122  0.0  0.0  0.0  0.0     43.0
60656.0    1018.0   4.636542    10.199848  0.0  0.0  1.0  5.0    138.0
60657.0   14746.0   8.663773   352.7

In [208]:
graffiti[graffiti["Response Time (Days)"] == 26186.0] # very bizzare result.

Creation Date     Status Completion Date Type of Service Request  \
671402    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671403    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671404    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671405    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671406    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671407    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671408    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671409    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671410    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671411    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671419    01/01/1944  Completed      09/11/2015        Graffiti Removal   
671420    01/01/1944  Completed      09/11/2015        Graffiti Removal   

       What Type of Surface is the Graffiti on?  \
671402     Cement (Sidewalk, Alley, Wall, Curb)   
671403     Cement (Sidewalk, Alley, Wall, Curb)   
671404                        Metal - Unpainted   
671405                        Metal - Unpainted   
671406                        Metal - Unpainted   
671407                        Metal - Unpainted   
671408                         Wood - Unpainted   
671409                        Brick - Unpainted   
671410                        Metal - Unpainted   
671411                        Metal - Unpainted   
671419     Cement (Sidewalk, Alley, Wall, Curb)   
671420     Cement (Sidewalk, Alley, Wall, Curb)   

       Where is the Graffiti located?  ZIP Code  Ward  Police District  \
671402                          Front   60621.0  20.0              7.0   
671403                          Alley   60629.0  23.0              8.0   
671404                           Sign   60632.0  23.0              8.0   
671405                           Sign   60629.0  13.0              8.0   
671406                           Sign   60629.0  13.0              8.0   
671407                           Rear   60632.0  14.0              8.0   
671408                          Fence   60632.0  14.0              8.0   
671409                           Rear   60632.0  14.0              8.0   
671410                           Sign   60609.0  12.0              9.0   
671411                           Sign   60609.0  12.0              9.0   
671419                          Front   60632.0  15.0              9.0   
671420                          Front   60632.0  15.0              9.0   

        Community Area  Response Time (Days)  
671402            68.0               26186.0  
671403            65.0               26186.0  
671404            57.0               26186.0  
671405            65.0               26186.0  
671406            65.0               26186.0  
671407            56.0               26186.0  
671408            56.0               26186.0  
671409            56.0               26186.0  
671410            59.0               26186.0  
671411            59.0               26186.0  
671419            58.0               26186.0  
671420            58.0               26186.0

In [209]:
graffiti['Response Time (Days)'].groupby([graffiti["ZIP Code"], graffiti['Where is the Graffiti located?']]).describe().unstack()

count       mean         std  min  \
ZIP Code Where is the Graffiti located?                                       
0.0      Alley                              4.0  14.000000   27.337398  0.0   
         Door                               4.0  19.250000   28.324018  0.0   
         Express Way Job                    4.0  16.000000   30.000000  1.0   
         Fence                              3.0   0.000000    0.000000  0.0   
         Front                             61.0   3.622951   18.265782  0.0   
         Garage                             5.0  39.000000   38.170669  0.0   
         Garbage Cart                       2.0  25.000000   35.355339  0.0   
         Mail Box                           4.0   0.000000    0.000000  0.0   
         Newspaper Box                      3.0   0.000000    0.000000  0.0   
         Pole                              23.0   0.608696    1.924463  0.0   
         Rear                               8.0  10.750000   18.468119  0.0   
         Side                              16.0  19.562500   35.141559  0.0   
         Sign                               4.0   0.250000    0.500000  0.0   
         Traffic Control Box               13.0  13.923077   37.444318  0.0   
60601.0  Alley                             46.0   3.608696    8.947944  0.0   
         Bench                             30.0   2.733333    5.271480  0.0   
         Door                             128.0   0.859375    2.486924  0.0   
         Dumpster                          84.0   0.261905    0.746382  0.0   
         Express Way Job                   32.0   2.906250    4.191846  0.0   
         Fence                             83.0   1.746988    3.869340  0.0   
         Front                           1986.0   1.078046    3.490101  0.0   
         Garage                            16.0   5.250000    8.029114  0.0   
         Garbage Cart                       8.0   5.750000    9.192388  0.0   
         Hydrant                            4.0   4.250000    5.188127  1.0   
         Mail Box                          19.0   0.684211    1.157230  0.0   
         Newspaper Box                     71.0   1.056338    5.497759  0.0   
         Overpass                          96.0   1.635417    3.299505  0.0   
         Pole                             433.0  12.145497  159.471638  0.0   
         Rear                              65.0   3.046154    4.570390  0.0   
         Side                             456.0   2.989035    5.727310  0.0   
...                                         ...        ...         ...  ...   
60666.0  Side                               2.0   6.000000    0.000000  6.0   
         Traffic Control Box                2.0   0.000000    0.000000  0.0   
60707.0  Alley                             79.0   3.518987    7.389981  0.0   
         Bench                             23.0   0.782609    1.204406  0.0   
         Door                              29.0   0.827586    1.833595  0.0   
         Dumpster                          37.0   0.189189    0.569495  0.0   
         Fence                             37.0   1.864865    4.613761  0.0   
         Front                            613.0   3.787928   18.385506  0.0   
         Garage                           313.0  15.875399  107.767766  0.0   
         Garbage Cart                      25.0   1.360000    3.289883  0.0   
         Hydrant                            2.0   3.000000    1.414214  2.0   
         Mail Box                          42.0   4.261905    8.505549  0.0   
         Newspaper Box                     41.0   0.926829    2.769388  0.0   
         Overpass                           3.0   0.333333    0.577350  0.0   
         Phone                              4.0   0.500000    0.577350  0.0   
         Pole                             229.0   0.484716    2.366799  0.0   
         Rear                             194.0   9.855670   32.163923  0.0   
         Side                             358.0   7.941341   11.780886  0.0   
         Sign

In [210]:
graffiti['Response Time (Days)'].groupby([graffiti["Where is the Graffiti located?"], graffiti['What Type of Surface is the Graffiti on?']]).describe().unstack()

count  \
Where is the Graffiti located? What Type of Surface is the Graffiti on?            
Alley                          Aluminum Siding                            2245.0   
                               Asphalt                                     135.0   
                               Brick - Painted                            6417.0   
                               Brick - Unpainted                         10830.0   
                               Cement (Sidewalk, Alley, Wall, Curb)       2655.0   
                               Glass                                        87.0   
                               Limestone                                    95.0   
                               Marble/Granite                               26.0   
                               Metal                                      1402.0   
                               Metal - Painted                            4473.0   
                               Metal - Unpainted                           510.0   
                               Other / Unknown Surface                    1781.0   
                               Other/Unknown Surface                       683.0   
                               Stucco                                      118.0   
                               Tree                                          9.0   
                               Vinyl Siding                               1066.0   
                               Wood - Painted                             1844.0   
                               Wood - Unpainted                            953.0   
Bench                          Aluminum Siding                               3.0   
                               Asphalt                                      11.0   
                               Brick - Painted                             109.0   
                               Brick - Unpainted                            40.0   
                               Cement (Sidewalk, Alley, Wall, Curb)        157.0   
                               Glass                                        61.0   
                               Limestone                                    16.0   
                               Marble/Granite                               11.0   
                               Metal                                        66.0   
                               Metal - Painted                             289.0   
                               Metal - Unpainted                            95.0   
                               Other / Unknown Surface                     212.0   
...                                                                          ...   
Traffic Control Box            Cement (Sidewalk, Alley, Wall, Curb)         68.0   
                               Glass                                         8.0   
                               Limestone                                     4.0   
                               Marble/Granite                                1.0   
                               Metal                                      6264.0   
                               Metal - Painted                           17045.0   
                               Metal - Unpainted                           346.0   
                               Other / Unknown Surface                    2579.0   
                               Other/Unknown Surface                        28.0   
                               Stucco                                        2.0   
                               Tree                                          5.0   
                               Vinyl Siding                                  2.0   
                               Wood - Painted                               44.0   
                               Wood - Unpainted                              2.0   
Viaduct                        Aluminum Siding                               4.0   
                               Asphalt     

In [211]:
buildings = pd.read_csv("311_Service_Requests_-_Vacant_and_Abandoned_Buildings_Reported.csv")

/Users/igabr/anaconda/lib/python3.5/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (4,10) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [212]:
buildings.shape

(58734, 23)

In [213]:
buildings["LOCATION OF BUILDING ON THE LOT (IF GARAGE, CHANGE TYPE CODE TO BGD)."].value_counts()

Front                                                                    42255
Rear                                                                      7233
Garage                                                                    1359
LOCATION OF BUILDING ON THE LOT (IF GARAGE, CHANGE TYPE CODE TO BGD).        1
Name: LOCATION OF BUILDING ON THE LOT (IF GARAGE, CHANGE TYPE CODE TO BGD)., dtype: int64

In [214]:
sum(pd.isnull(buildings['LOCATION OF BUILDING ON THE LOT (IF GARAGE, CHANGE TYPE CODE TO BGD).']))

7886

In [215]:
buildings.columns

Index(['SERVICE REQUEST TYPE', 'SERVICE REQUEST NUMBER',
       'DATE SERVICE REQUEST WAS RECEIVED',
       'LOCATION OF BUILDING ON THE LOT (IF GARAGE, CHANGE TYPE CODE TO BGD).',
       'IS THE BUILDING DANGEROUS OR HAZARDOUS?',
       'IS BUILDING OPEN OR BOARDED?',
       'IF THE BUILDING IS OPEN, WHERE IS THE ENTRY POINT?',
       'IS THE BUILDING CURRENTLY VACANT OR OCCUPIED?',
       'IS THE BUILDING VACANT DUE TO FIRE?',
       'ANY PEOPLE USING PROPERTY? (HOMELESS, CHILDEN, GANGS)',
       'ADDRESS STREET NUMBER', 'ADDRESS STREET DIRECTION',
       'ADDRESS STREET NAME', 'ADDRESS STREET SUFFIX', 'ZIP CODE',
       'X COORDINATE', 'Y COORDINATE', 'Ward', 'Police District',
       'Community Area', 'LATITUDE', 'LONGITUDE', 'Location'],
      dtype='object')

In [216]:
potholes = pd.read_csv("311_Service_Requests_-_Pot_Holes_Reported.csv")

In [217]:
potholes

CREATION DATE           STATUS COMPLETION DATE SERVICE REQUEST NUMBER  \
0         01/01/2011  Completed - Dup      01/03/2011            11-00002110   
1         01/01/2011  Completed - Dup      01/03/2011            11-00002209   
2         01/01/2011  Completed - Dup      01/03/2011            11-00002224   
3         01/01/2011  Completed - Dup      01/03/2011            11-00002247   
4         01/01/2011  Completed - Dup      01/03/2011            11-00002276   
5         01/01/2011  Completed - Dup      01/03/2011            11-00002947   
6         01/01/2011  Completed - Dup      01/03/2011            11-00003041   
7         01/01/2011  Completed - Dup      01/03/2011            11-00003086   
8         01/01/2011  Completed - Dup      01/03/2011            11-00003439   
9         01/01/2011        Completed      01/03/2011            11-00002021   
10        01/01/2011        Completed      01/03/2011            11-00002273   
11        01/01/2011        Completed      01/03/2011            11-00002650   
12        01/01/2011        Completed      01/03/2011            11-00002654   
13        01/01/2011        Completed      01/03/2011            11-00002656   
14        01/01/2011        Completed      01/03/2011            11-00002913   
15        01/01/2011        Completed      01/03/2011            11-00002927   
16        01/01/2011        Completed      01/03/2011            11-00003121   
17        01/01/2011        Completed      01/03/2011            11-00003356   
18        01/01/2011        Completed      01/03/2011            11-00003428   
19        01/01/2011        Completed      01/04/2011            11-00001252   
20        01/01/2011        Completed      01/04/2011            11-00001597   
21        01/01/2011        Completed      01/04/2011            11-00001728   
22        01/01/2011        Completed      01/04/2011            11-00002193   
23        01/01/2011        Completed      01/04/2011            11-00002392   
24        01/01/2011        Completed      01/04/2011            11-00002825   
25        01/01/2011        Completed      01/04/2011            11-00002863   
26        01/01/2011        Completed      01/04/2011            11-00003132   
27        01/01/2011        Completed      01/04/2011            11-00003153   
28        01/01/2011        Completed      01/05/2011            11-00001242   
29        01/01/2011        Completed      01/05/2011            11-00001445   
...              ...              ...             ...                    ...   
457841    01/29/2016       Open - Dup             NaN            16-00580842   
457842    01/29/2016       Open - Dup             NaN            16-00583375   
457843    01/29/2017       Open - Dup             NaN            17-00527944   
457844    01/31/2017       Open - Dup             NaN            17-00575213   
457845    02/03/2017       Open - Dup             NaN            17-00631775   
457846    02/03/2017             Open             NaN            17-00624003   
457847    02/04/2017             Open             NaN            17-00644648   
457848    02/05/2016       Open - Dup             NaN            16-00722236   
457849    02/06/2017       Open - Dup             NaN            17-00668137   
457850    02/08/2017             Open             NaN            17-00719948   
457851    02/11/2016       Open - Dup             NaN            16-00853166   
457852    02/11/2017       Open - Dup             NaN            17-00770536   
457853    02/12/2017       Open - Dup             NaN            17-00779442   
457854    02/13/2017       Open - Dup             NaN            17-00804169   
457855    02/13/2017       Open - Dup             NaN            17-00804181   
457856    02/13/2017       Open - Dup             NaN            17-00804193   
457857    02/13/2017       Open - Dup             NaN            17-00804227   
457858    02/13/2017       Open - Dup             NaN            17-00804240   
457859 

In [218]:
sanitation = pd.read_csv("311_Service_Requests_-_Sanitation_Code_Complaints.csv")

In [219]:
sanitation.shape

(118827, 16)

In [220]:
# sanitation.columns = ['ZIP_CODE' if x=='ZIP Code' else x for x in sanitation.columns]

In [221]:
# buildings.columns = ['ZIP_CODE' if x=='ZIP CODE' else x for x in buildings.columns]

In [222]:
# pd.merge(sanitation, buildings, on='ZIP_CODE', how='inner')